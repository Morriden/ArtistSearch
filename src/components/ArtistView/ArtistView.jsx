import React from 'react';
import PropTypes from 'prop-types';

const ArtistView = ({ releases }) => {
  const artistReleases = releases.map(release => (
    <li key={release.id}>
      <div>
        {release.title}, {release.date}
        <img src={`http://coverartarchive.org/release/${release.id}/front`}/>
      </div>
    </li>
  ));

  return (
    <section>
      {artistReleases}
    </section>   
  );
};

ArtistView.propTypes = {
  releases: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default ArtistView;
