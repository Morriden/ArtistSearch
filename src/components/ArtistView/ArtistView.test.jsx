import React from 'react';
import { shallow } from 'enzyme';
import ArtistView from './ArtistView';

describe('Displays Releases for and Artist', () => {
  it('Should be the releases from an Artist', () => {
    const wrapper = shallow(<ArtistView releases={[{
      title: 'title1', date: 'somedate', id: 'ID1'
    },
    {
      title: 'title2', date: 'somedate', id: 'ID2'
    }, {
      title: 'title3', date: 'somedate', id: 'ID3'
    }]}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
