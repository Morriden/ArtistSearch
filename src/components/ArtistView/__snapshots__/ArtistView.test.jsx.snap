// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays Releases for and Artist Should be the releases from an Artist 1`] = `
<section>
  <li
    key="ID1"
  >
    <div>
      title1
      , 
      somedate
      <img
        src="http://coverartarchive.org/release/ID1/front"
      />
    </div>
  </li>
  <li
    key="ID2"
  >
    <div>
      title2
      , 
      somedate
      <img
        src="http://coverartarchive.org/release/ID2/front"
      />
    </div>
  </li>
  <li
    key="ID3"
  >
    <div>
      title3
      , 
      somedate
      <img
        src="http://coverartarchive.org/release/ID3/front"
      />
    </div>
  </li>
</section>
`;
